src/Builder/FrontendDialectHelper.cpp

```diff
@@ -269,10 +269,12 @@ mlir::Type convertONNXTypeToMLIRType(
   case onnx::TensorProto_DataType::TensorProto_DataType_COMPLEX64:
   case onnx::TensorProto_DataType::TensorProto_DataType_COMPLEX128:
   case onnx::TensorProto_DataType::TensorProto_DataType_UNDEFINED:
-  default:
     assert(false && "Unsupported data type encountered.");
     return nullptr;
   }
+
```

Comment:
I believe the preferred option for such case in LLVM compilers is to use `    llvm_unreachable("unexpected type");` The LLVM compiler will know not to expect a return after an unreachable.

Reply:
makes sense. i'll make the change